src/Graphing.hs

```diff
@@ -396,26 +396,45 @@ subGraphOf n (Graphing gr) =
     keepPredicate Root = True
     keepPredicate (Node ty) = Set.member (Node ty) reachableNodes
 
--- | Adds color to all nodesToColor, with the color being 'a's extractedProperty
+-- | Coloring a graph allows you to attach arbitrary context to nodes.
+--   For example, we can attach `color` to graph nodes like this:
+
+--   Given a graph of type:
 --
--- data SimpleDep = SimpleDep{ name :: Text, colorTag :: Text, colors :: Set Text}
+-- >    data SimpleDep = SimpleDep
+-- >      { name :: Text
+-- >      , colorTag :: Text
+-- >      , colors :: Set Text
+-- >      }
 --
--- updateColors :: Set Text -> SimpleDep -> SimpleDep
--- updateColors newColor dep = dep{colors = newColor}
-
--- extractColors :: SimpleDep -> Set Text
--- extractColors = colors
-
--- extractColorTag :: SimpleDep -> Text
--- extractColorTag = colorTag
-
--- Example
+--   Where the graph is structured as:  d1 -> d2 -> d3
+--
+-- >    d1 = SimpleDep              d2 = SimpleDep                 d3 = SimpleDep
+-- >      { name : "d1"               { name : "d2"                   { name : "d3"
+-- >      , colorTag : "blue"         , colorTag : "green"            , colorTag : "red"
+-- >      , colors : mempty           , colors : mempty               , colors : mempty
+-- >      }                           }                               }
 --
---  d1("d1", "blue" mempty) -> d2("d2", "green" mempty) -> d3("d3", "red" mempty)
+-- > extractColorTag :: SimpleDep -> Text
+-- > extractColorTag = colorTag
 --
--- color gr extractColors updateColors d1 extractColorTag nodesToColor= Set.fromList["blue", "green", "red"] =
+-- > extractColors :: SimpleDep -> Set Text
+-- > extractColors = colors
 --
--- d1("d1", "blue" Set("blue")) -> d2("d2", "green" Set("blue")) -> d3("d3", "red" Set("blue"))
+-- > updateColors :: Set Text -> SimpleDep -> SimpleDep
+-- > updateColors newColor dep = dep{colors = newColor
+--
+-- > color gr extractColors updateColors d1 extractColorTag nodesToColor = Set.fromList["green", "red"]
+--
+--   d2 and d3 after computation:
+-- >    d2 = SimpleDep                 d3 = SimpleDep
+-- >      { name : "d2"                   { name : "d3"
+-- >      , colorTag : "green"            , colorTag : "red"
+-- >      , colors : Set("blue")          , colors : Set("blue")
+-- >      }                               }
+--
+--   In the context Maven submodule filtering (Maven/Common.hs), `color` is used to attach submodule information to nodes.
+--   This is later used to determine which nodes to filter based on the provided submodule filters.
```

Comment:
Avoid coupling an explanation of this function (which is generalized to all `Ord a`) to uses of the functions which are more specialized. A reader should be able to understand how `color` works without having to think about how Maven specifically uses it. If this is just an example for illustrative purposes then I guess it's OK if you make it clearer that it's an example.

Recommendation: This information is more relevant and useful in `Maven/Common.hs`. Please move or delete it.